prov/efa/src/efa_base_ep.h

```diff
@@ -39,6 +39,26 @@
 #define efa_rx_flags(efa_base_ep) ((efa_base_ep)->util_ep.rx_op_flags)
 #define efa_tx_flags(efa_base_ep) ((efa_base_ep)->util_ep.tx_op_flags)
 
+#define EFA_PROV_NAME "efa"
+#define EFA_DIRECT_PROV_NAME "efa-direct"
+
+#define EFA_INFO_TYPE_IS_RDM(_info) \
+	(_info && _info->ep_attr && (_info->ep_attr->type == FI_EP_RDM) && !strcasecmp(_info->fabric_attr->name, EFA_PROV_NAME))
```

Comment:
Are the linear string comparisons necessary? These are already pretty gnarly chained conditionals.

If the `fi_info.fabric_attr.name` is truly the only way to disambiguate efa-rdm from efa-direct (I.e. we can't add some other enum field elsewhere), I'd like to explore employing constant-time comparison schemes.

Reply:
Is `strncmp` with the length of the longer string good enough? These macros are only used in the `fi_getinfo` and `fi_domain` calls, so they are not performance critical.